xs/src/libslic3r/PerimeterGenerator.cpp

```diff
@@ -54,7 +71,7 @@ PerimeterGenerator::process()
     for (Surfaces::const_iterator surface = this->slices->surfaces.begin();
         surface != this->slices->surfaces.end(); ++surface) {
         // detect how many perimeters must be generated for this island
-        const int loop_number = this->config->perimeters + surface->extra_perimeters -1;  // 0-indexed loops
+        const int loop_number = num_loops(this->config->perimeters, surface->extra_perimeters);//this->config->perimeters + surface->extra_perimeters-1;  // 0-indexed loops
```

Comment:
loop_number is a constant. Any preference on how to set it? I thought a new method would be okay but you do bring up a good point.